xinference/core/model.py

```diff
@@ -110,6 +110,9 @@ def load(self):
 
     async def _wrap_generator(self, ret: Any):
         if inspect.isgenerator(ret):
+            if self._lock is not None:
+                # Make sure only one iterator is valid.
+                self._generators.clear()
```

Comment:
How about simply raise an exception when self._lock is not None and self._generators is not empty?

Reply:
Current behavior similars to C++ iterator, append data invalidate old iterators.

Reply:
Fixed. Thanks.